### DISCUSSION TITLE
Why is k3s creating /tmp/k3s-resolve.conf with nameserver 8.8.8.8?

### DISCUSSION THREAD

**Discussion by the-scott-hand** [k3s-io/k3s#3867] [Q&A] [answered]
Context:
I am spinning up a cluster using k3d. Inside the container is a file `/etc/resolv.conf`. Also inside the contain is a file `/tmp/k3s-resolv.conf` with the contents `nameserver: 8.8.8.8`. I am running in an air gapped network and do not want to reach out to 8.8.8.8. 

Why is `/tmp/k3s-resolv.conf` being created? How can I avoid it being created/used? From what I can tell if I have a valid /etc/resolv.conf this file should not be made as per https://github.com/k3s-io/k3s/blob/master/pkg/agent/config/config.go#L312-L328 (which maybe i am not fully understanding?).

**Comment by ChristianCiach** [comment, marked as answer]
I found this comment: https://github.com/k3s-io/k3s/issues/53#issuecomment-476908192

> The settings are configurable in that we will either take a --resolv-conf flag to pass down to the kubelet, or a K3S_RESOLV_CONF environment variable will work also. We now try to use system resolv.conf files (from /etc & systemd), and will create a /tmp/k3s-resolv.conf file with nameserver 8.8.8.8 if nameservers in the system files are not global unicast ips.

You can provide your own resolv.conf to K3s, but I don't know how to do that when using K3d.

**Reply by brandond** [reply]
If you have IPv6 resolvers in your resolv.conf, k3s will generate its own using 8.8.8.8, as using IPv6 upstream DNS servers will break DNS in pods unless you have dual-stack networking. You might go ask in the K3d repo what's the best way to use your own resolv.conf.

**Reply by the-scott-hand** [reply]
That makes sense, I am beginning to suspect that the default provided /etc/resolv.conf in the container is considered invalid by the code i posted. Thanks!
